sign-plugin/src/main/java/eu/cloudnetservice/sign/plugin/hook/ThreadImpl.java

```diff
@@ -0,0 +1,223 @@
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.util.Vector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ThreadImpl extends Thread {
+
+	private final SignNetworkHandlerAdapter signNetworkHandlerAdapter;
+
+	int animationTick = 1;
+	private boolean valueTick = false;
+
+	public ThreadImpl(SignNetworkHandlerAdapter signNetworkHandlerAdapter) {
+		this.signNetworkHandlerAdapter = signNetworkHandlerAdapter;
+	}
+
+	@Override
+	public void run() {
+		while (!isInterrupted()) {
+			if (SignManager.getInstance().getSignLayoutConfig() != null && SignManager.getInstance().getSignLayoutConfig().isKnockbackOnSmallDistance()) {
+				try {
+					for (Sign sign : SignManager.getInstance().getSigns().values()) {
+						if (Bukkit.getWorld(sign.getPosition().getWorld()) != null) {
+							Location location = signNetworkHandlerAdapter.toLocation(sign.getPosition());
+							for (Entity entity : location.getWorld().getNearbyEntities(location,
+									SignManager.getInstance().getSignLayoutConfig().getDistance(),
+									SignManager.getInstance().getSignLayoutConfig().getDistance(),
+									SignManager.getInstance().getSignLayoutConfig().getDistance())) {
+								if (entity instanceof Player && !entity.hasPermission("cloudnet.signs.knockback.bypass")) {
+									Bukkit.getScheduler().runTask(signNetworkHandlerAdapter.getPlugin(), new Runnable() {
+										@Override
+										public void run() {
+											if (location.getBlock().getState() instanceof org.bukkit.block.Sign) {
+												try {
+													Location entityLocation = entity.getLocation();
+													entity.setVelocity(new Vector(entityLocation.getX() - location.getX(),
+															entityLocation.getY() - location.getY(),
+															entityLocation.getZ() - location.getZ()).normalize()
+															.multiply(
+																	SignManager.getInstance().getSignLayoutConfig()
+																			.getStrength())
+															.setY(0.2D));
+												} catch (Exception ex) {
+													ex.printStackTrace();
+												}
+											}
+										}
+									});
+								}
+							}
+						}
+					}
+				} catch (Exception ex) {
+
+				}
+			}
+
+			if (SignManager.getInstance().getSignLayoutConfig() != null) {
+
+				SearchingAnimation searchingAnimation = SignManager.getInstance().getSignLayoutConfig().getSearchingAnimation();
+
+				SignLayout searchLayer = signNetworkHandlerAdapter.getSearchingLayout(animationTick);
+				Bukkit.getScheduler().runTask(signNetworkHandlerAdapter.getPlugin(), () -> {
+					for (Sign sign : SignManager.getInstance().getSigns().values()) {
+						boolean exists = signNetworkHandlerAdapter.exists(sign);
+
+						if (!exists) {
+							sign.setServerInfo(null);
+							continue;
+						}
+
+						if (signNetworkHandlerAdapter.isMaintenance(sign.getTargetGroup())) {
+							SignLayout _signLayout = signNetworkHandlerAdapter.getLayout(sign.getTargetGroup(), "maintenance");
```

Comment:
this name doesn't say anything, why it is special. what does the underscore even mean?